### DISCUSSION TITLE
Update authstore data

### DISCUSSION THREAD

**Discussion by ttijl** [pocketbase/pocketbase#2600] [Q&A] [answered]
Hello, i currently have this code for the client side.

```ts
import PocketBase from 'pocketbase';

import { writable } from 'svelte/store';

export const pb = new PocketBase('http://127.0.0.1:8090');

export const currentUser = writable(pb.authStore.model);

pb.authStore.onChange((auth) => {
    currentUser.set(pb.authStore.model);
});
```

Even if you refresh the page it still keeps the userdata from the first login. How can i make the userdata update in realtime or update it manually?

**Comment by ganigeorgiev** [comment, marked as answer]
By default the authStore will keep (and load) the previously authenticated model and token in LocalStorage.

If you want to get an up-to-date user data (and token with renewed `exp` claim), you can call `authRefresh()`:

```js
// on success - automatically updates the pb.authStore
// otherwise throws an error on invalid or expired pb.authStore.token
await pb.collection("users").authRefresh();
```

Alternatively, you can also subscribe to the realtime events of your current user, something like:

```js
pb.collection("users").subscribe(pb.authStore.model.id, (e) => {
    // both operations will trigger the authStore.onChange listener
    if (e.action == "delete") {
        pb.authStore.clear();
    } else {
        pb.authStore.save(pb.authStore.token, e.record);
    }
});
```

**Reply by rubenuijtdewilligen** [reply]
Thank you, this reply also helped me. One question, does calling
```JS
await pb.collection("users").authRefresh();
```
in the handle hook put a lot more load on the server PocketBase is running on? Or is it fine for performance? My app is used in sessions of about 5 minutes every week by 200 users, so it's not too large if that makes a difference.

**Reply by ganigeorgiev** [reply]
@rubenuijtdewilligen It is OK to call `authRefresh()` on every page load/refresh, if that is what you are asking. Based on the [benchmarks](https://github.com/pocketbase/benchmarks/blob/master/results/hetzner_cax11.md#user-auth-refresh) even on the smallest Hetzner VPS  `authRefresh()` is just as fast as `getOne()` (_which is practically a direct fetch by primary key when there is no extra View API rule, aka. the most optimized query you could get_).
